plugins/pencilblue/templates/angular/admin/content/articles/article_form.html

```diff
@@ -35,6 +36,21 @@
 			});
 		};
 
+		$scope.generateUrl = function() {
+			$scope.article.url = $scope.article.headline.replace(/\s+/g, '-').toLowerCase();
+			if($scope.article.url){
+				$scope.urlGenerated = true;
+			} else {
```

Comment:
We'll never have this hit, if the headline is valid. Also, I would like to make sure the URL is unique/valid. You can make a call to the check function at this point to make sure of that.